### ISSUE TITLE
Encapsulate Complex Queries: Combine requests

### ISSUE THREAD

**Issue by tobiasschweizer** [dhlab-basel/knora-api-js-lib#26]
When doing a full resource requests, the incoming links should be queried as well (this could be made optional if the client is not interested in it). Similarly, some media types require more information than the full resource requests returns, see https://github.com/dhlab-basel/knora-api-js-lib/issues/5#issue-465242546.

In these cases, the result of the full resource request should be combined with the result of a Gravsearch query that gets the additional information. 

RxJS's `withLatestFrom` (https://rxjs-dev.firebaseapp.com/api/operators/withLatestFrom) could be used. The full resource request would act as the guiding stream and be combined with the result of the Gravsearch query, emitting one combined result to the client.

see https://blog.angularindepth.com/learn-to-combine-rxjs-sequences-with-super-intuitive-interactive-diagrams-20fce8e6511

**Comment by tobiasschweizer** [comment]
it would be nice if we could return the result of the full resource requests immediately when available, also when the Gravsearch result is not there yet. Once it is there, the client would receive another call to `next`.

**Comment by tobiasschweizer** [comment]
@andreas-aeschlimann Do yo have an idea how to do this (https://github.com/dhlab-basel/knora-api-js-lib/issues/26#issuecomment-529828246)?

**Comment by tobiasschweizer** [comment]
could be useful: https://stackoverflow.com/questions/54039425/rxjs-combinelatest-without-waiting-for-source-observales-to-emit

**Comment by tobiasschweizer** [comment]
after discussion with @lrosenth:

We need to be able to handle paging when dealing with the search endpoint. Idea: the client could use a class that performs the first page of a search result and offers a `next()` method that can be called to get the next page, integrating the new results into the same structure.

**Comment by tobiasschweizer** [comment]
see https://dev.to/exacs/es2018-real-life-simple-usage-of-async-iteration-get-paginated-data-from-rest-apis-3i2e

**Comment by tobiasschweizer** [comment]
async iterators: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/asyncIterator

**Comment by kilchenmann** [comment]
Any progress on this issue? I saw that incomingReferences and outgoingReferences are always empty in the ReadResource object...

**Comment by tobiasschweizer** [comment]
> Any progress on this issue? I saw that incomingReferences and outgoingReferences are always empty in the ReadResource object...

It depends on what your request looks like. If you do a Gravsearch query, you could get incoming and outgoing refs.

**Comment by kilchenmann** [comment]
No, only in the getResource request. I thought I will get the pages in resource 'book' in incomingReferences? Or do I have to do an additional request?

**Comment by tobiasschweizer** [comment]
>  I thought I will get the pages in resource 'book' in incomingReferences?

No, this is not how it works. You have to do a Gravsearch request to get incoming references.
